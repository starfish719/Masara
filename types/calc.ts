export interface CalcResult {
  effortValue: number;
  resultValue: number;
}
