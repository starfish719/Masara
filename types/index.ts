export * from './pokemon';
export * from './calc';
export * from './speed';
