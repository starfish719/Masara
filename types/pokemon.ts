export interface Pokemon {
  id: number;
  name: string;
  baseStatsS: number;
}

export interface SelectPokemon {
  value: string;
  label: string;
  baseStats: number;
}
