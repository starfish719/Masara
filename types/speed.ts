import { SelectPokemon } from './';

export interface SpeedMaster {
  calcResult: number;
  description: {
    ja: string;
  };
}

export interface CalcProps {
  pokemon: SelectPokemon;
  effortValue: number;
  individualValue: number;
  personality: number;
  rank: number;
  item: number;
  ability: number;
  isTailwind: boolean;
  isParalysis: boolean;
}
