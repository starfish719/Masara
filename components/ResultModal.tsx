import { CalcResult } from '../types';
import { getResultDescription } from '../models/SpeedMaster';
import Modal from 'react-modal';

type Props = {
  isOpen: boolean;
  calcResult: CalcResult[];
  onClose: () => void;
};

export const ResultModal = (props: Props) => {
  const { isOpen, calcResult, onClose } = props;

  const customStyles = {
    content: {
      border: 'none',
      background: 'none',
      width: '100%',
      inset: 0
    },
  };

  return (
    <Modal
      ariaHideApp={false}
      isOpen={isOpen}
      contentLabel="CalcResult"
      style={customStyles}
    >
      <div className="w-full items-start min-h-screen pt-4 px-4 pb-20 text-left sm:block sm:p-0">
        <span
          className="hidden sm:inline-block sm:align-middle sm:h-screen"
          aria-hidden="true"
        >
          &#8203;
        </span>
        <div className="w-full inline-block align-bottom bg-white rounded-lg overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle">
          <div className="bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4 min-w-full">
            <div className="sm:flex sm:items-start">
              <div className="mt-3 sm:mt-0 sm:text-left min-w-full">
                <div
                  className="mt-2 overflow-y-auto"
                  style={{ maxHeight: '70vh' }}
                >
                  {calcResult.map((result) => {
                    return (
                      <div
                        key={result.effortValue}
                        className="pt-3 pb-3 border-b-2"
                      >
                        <p className="text-sm text-gray-500 mb-1">
                          実数値: {result.resultValue}
                        </p>
                        <p className="text-sm text-gray-500 mb-1">
                          努力値: {result.effortValue}
                        </p>
                        <p className="text-sm text-gray-500">
                          {getResultDescription(result.resultValue)}
                        </p>
                      </div>
                    );
                  })}
                </div>
              </div>
            </div>
          </div>
          <div className="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse">
            <button
              type="button"
              className="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:text-sm"
              onClick={onClose}
            >
              Close
            </button>
          </div>
        </div>
      </div>
    </Modal>
  );
};
