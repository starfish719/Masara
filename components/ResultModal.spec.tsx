import { fireEvent, render } from '@testing-library/react';

import { CalcResult } from '../types';
import { ResultModal } from './ResultModal';

describe('ResultModal', () => {
  const onCloseMock = jest.fn();
  const calcResultMock: CalcResult[] = [
    {
      effortValue: 0,
      resultValue: 100,
    },
    {
      effortValue: 4,
      resultValue: 101,
    },
    {
      effortValue: 8,
      resultValue: 102,
    },
  ];

  it('when isOpen is false, should render', () => {
    const { asFragment } = render(
      <ResultModal
        isOpen={false}
        calcResult={calcResultMock}
        onClose={onCloseMock}
      />,
      { container: document.body },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('when isOpen is true, should render', () => {
    const { asFragment } = render(
      <ResultModal
        isOpen={true}
        calcResult={calcResultMock}
        onClose={onCloseMock}
      />,
      { container: document.body },
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('when clicked close button, call onClose', () => {
    const { getByText } = render(
      <ResultModal
        isOpen={true}
        calcResult={calcResultMock}
        onClose={onCloseMock}
      />,
      { container: document.body },
    );

    fireEvent.click(getByText('Close'));
    expect(onCloseMock).toBeCalled();
  });
});
