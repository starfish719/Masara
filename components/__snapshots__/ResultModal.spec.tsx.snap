// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultModal when isOpen is false, should render 1`] = `
<DocumentFragment>
  <div
    class="ReactModalPortal"
  />
</DocumentFragment>
`;

exports[`ResultModal when isOpen is true, should render 1`] = `
<DocumentFragment>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
    >
      <div
        aria-label="CalcResult"
        aria-modal="true"
        class="ReactModal__Content"
        role="dialog"
        style="position: absolute; top: 40px; left: 40px; right: 40px; bottom: 40px; background: none; overflow: auto; border-radius: 4px; outline: none; padding: 20px; width: 100%; inset: 0;"
        tabindex="-1"
      >
        <div
          class="w-full items-start min-h-screen pt-4 px-4 pb-20 text-left sm:block sm:p-0"
        >
          <span
            aria-hidden="true"
            class="hidden sm:inline-block sm:align-middle sm:h-screen"
          >
            ​
          </span>
          <div
            class="w-full inline-block align-bottom bg-white rounded-lg overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle"
          >
            <div
              class="bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4 min-w-full"
            >
              <div
                class="sm:flex sm:items-start"
              >
                <div
                  class="mt-3 sm:mt-0 sm:text-left min-w-full"
                >
                  <div
                    class="mt-2 overflow-y-auto"
                    style="max-height: 70vh;"
                  >
                    <div
                      class="pt-3 pb-3 border-b-2"
                    >
                      <p
                        class="text-sm text-gray-500 mb-1"
                      >
                        Actual: 100
                      </p>
                      <p
                        class="text-sm text-gray-500 mb-1"
                      >
                        Effort: 0
                      </p>
                      <p
                        class="text-sm text-gray-500"
                      >
                        無振り80族（トゲキッス/フシギバナ/カブトプス/カイリュー/バシャーモ/ダーテング/サーナイト/チルタリス/オニゴーリ/フワライド/マンムー/エルレイド/ムーランド/シャンデラ/バルジーナ/ウォーグル/ヌメルゴン/ナゲツケサル/レジドラゴ/バドレックス）
                      </p>
                    </div>
                    <div
                      class="pt-3 pb-3 border-b-2"
                    >
                      <p
                        class="text-sm text-gray-500 mb-1"
                      >
                        Actual: 101
                      </p>
                      <p
                        class="text-sm text-gray-500 mb-1"
                      >
                        Effort: 4
                      </p>
                      <p
                        class="text-sm text-gray-500"
                      >
                        最速40族（ヌケニン/ドサイドン）無振り81族（ギャラドス/ミロカロス）
                      </p>
                    </div>
                    <div
                      class="pt-3 pb-3 border-b-2"
                    >
                      <p
                        class="text-sm text-gray-500 mb-1"
                      >
                        Actual: 102
                      </p>
                      <p
                        class="text-sm text-gray-500 mb-1"
                      >
                        Effort: 8
                      </p>
                      <p
                        class="text-sm text-gray-500"
                      >
                        準速50族（ラフレシア/ラッキー/キレイハナ/ハピナス/ヤミラミ/ツンベアー/メレシー/バドレックス白）無振り82族（ルガルガン夜）
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse"
            >
              <button
                class="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:text-sm"
                type="button"
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
